src/ndarray/ndarray.cc

```diff
@@ -1214,6 +1214,40 @@ void NDArray::SyncCopyToCPU(void *data, size_t size) const {
   }
 }
 
+void NDArray::CheckFormat(const bool full_check) const {
+  int32_t err = kNormalErr;
+  TBlob err_cpu(&err, mshadow::Shape1(1), cpu::kDevMask, 0);
+  if (this->ctx().dev_mask() == cpu::kDevMask) {
+    Engine::Get()->PushSync([&](RunContext rctx) {
+        common::CheckFormatWrapper<cpu>(rctx, *this, err_cpu, full_check);
+      }, this->ctx(), {this->var()}, {},
+      FnProperty::kNormal, 0, PROFILER_MESSAGE("CheckFormat"));
+    this->WaitToWrite();
+  } else {
+#if MXNET_USE_CUDA
+    Engine::Get()->PushSync([&](RunContext rctx) {
+        common::CheckFormatWrapper<gpu>(rctx, *this, err_cpu, full_check);
+        rctx.get_stream<gpu>()->Wait();
+      }, this->ctx(), {this->var()}, {},
+      FnProperty::kNormal, 0, PROFILER_MESSAGE("CheckFormat"));
+    this->WaitToWrite();
+#else
+    LOG(FATAL) << "GPU is not enabled";
+#endif
+  }
+  CHECK_NE(err, kCSRShapeErr) << "Shape mismatch of this CSRNDArray";
+  CHECK_NE(err, kCSRIndPtrErr)
+           << "IndPtr of CSRNDArray should be in non-decreasing order, start with 0, "
+           << "and end with value greater or equal than size of indices.";
```

Comment:
Also mention that indices should not be negative for csr & rsp. For csr indices should be sorted per row.